### PR TITLE
Add `isLocal` & `delete..` lightning service methods

### PR DESCRIPTION
**Pull request by vindard**
## Description

Add two new needed lightning service methods to codebase:
- `isLocal`
- `deleteUnpaidInvoice`

### CODE REVIEW

**Review by bodymindarts — APPROVED** [review]

File: src/core/lightning/index.ts [hunk 1]
```diff
@@ -318,9 +323,13 @@ export const LightningMixin = (superclass) =>
             // if we failed to do it, the invoice would still be present in InvoiceUser
             // in case the invoice were to be paid another time independantly (unlikely outcome)
             try {
-              const { lnd } = getLndFromPubkey({ pubkey })
-
-              await cancelHodlInvoice({ lnd, id })
+              const lndService = LndService()
+              if (lndService instanceof Error) return lndService
+              const deleteResult = lndService.deleteUnpaidInvoice({
```
Comment (nicolasburtey) [inline src/core/lightning/index.ts:328]: I think Unpaid in `deleteUnpaidInvoice` may be a bit misleading. the invoice was paid if we are in this branch. just not through lnd but with intra ledger.

Comment (vindard) [reply src/core/lightning/index.ts:328]: Ok I can maybe use something like `cancelInvoice` instead?

Comment (vindard) [reply src/core/lightning/index.ts]: Done here: https://github.com/GaloyMoney/galoy/pull/581/commits/99d27842d1cd175ebee5580928804bde6e5bf0ca

**Review by bodymindarts — APPROVED** [review]

**Review by bodymindarts — APPROVED** [review]

**Review by samerbuna — APPROVED** [review]
💪🏽 💪🏽 💪🏽  So much more readable

**Pull request merged**